app/controllers/account_reset/cancel_controller.rb

```diff
@@ -1,8 +1,9 @@
 module AccountReset
   class CancelController < ApplicationController
     def cancel
-      if AccountResetService.cancel_request(params[:token])
-        handle_success
+      account_reset = AccountResetService.cancel_request(params[:token])
```

Comment:
Can we please follow the convention where the Service/Form that the controller interacts with returns a FormResponse object? That way, you only need to make one call to analytics (as opposed to one for success, and one for failure), and it makes things consistent.

Reply:
In other words, the User UUID would be captured as `extra_analytics_attributes` inside the Service Object. Here's an example: https://github.com/18F/identity-idp/blob/master/app/services/email_confirmation_token_validator.rb

Reply:
As would the event name and token_valid attributes. The only parameter that would go in the analytics call is `result.to_h`, like we do everywhere else.

Reply:
While I agree with that convention, the complexity of this method does not warrant an additional object.

Reply:
Another test I like to do is to try to describe what the class does in one sentence. If the sentence contains the word `and`, that's usually a sign it's doing too much. For example, the current `AccountResetService` could be described as "it handles account reset requests, and account reset cancellations, and reports fraud, and sends notifications." That's a lot of stuff for one class!

Reply:
That's not to say that the one public method the class has can only do one thing. The class can make calls to other services, as long as those other calls are to private methods, and usually those methods would then call other classes to perform those actions. For example:
```ruby
class CancelAccountReset
  def submit(params)
    success = valid?
    if success
      notify_user_via_email 
      notify_user_via_sms
    end
  end
  
  private
  
  def notify_user_via_email
    UserMailer.send_email
  end
  
  def notify_user_via_sms
    SmsAccountResetCancellationNotifierJob.perform_now
  end
end
```

Reply:
You misunderstood.  I love breaking up the service object.  In fact I used that same methodology on controllers to breakup the rails monolith and everything funneling through routes.  What I was referring to was an additional response object.

Reply:
The complexity of the method should not determine whether or not an additional response object is necessary. It's about consistency and expectations. We expect all analytics events to have at a minimum `success` and `errors` attributes so we can easily differentiate between successful and unsuccessful events. This makes queries consistent for people searching in Kibana, or for the analytics team to run queries and monitor events. If I wanted to see what percentage of cancellations had an invalid token, I would normally look for `properties.event_properties.success`, but that doesn't exist in this case. So then, I would have to dig into all the properties and figure out which key to use to determine success, which in this case is a custom key called `token_valid`. This results in a poor developer experience.